cycode/cli/printers/table_printer.py

```diff
@@ -1,170 +1,168 @@
-                title = "Dependencies Vulnerabilities"
+        click.echo(text_table.draw())
 
-                headers = [SEVERITY_COLUMN] + headers
-                headers.extend(PREVIEW_DETECTIONS_COMMON_HEADERS)
-                headers.append(CVE_COLUMN)
-                headers.append(UPGRADE_COLUMN)
+    def _is_git_repository(self) -> bool:
+        return self.context.obj.get('remote_url') is not None
 
-                for detection in detections:
-                    rows.append(self._get_upgrade_package_vulnerability(detection))
-            elif detection_type_id == LICENSE_COMPLIANCE_POLICY_ID:
-                title = "License Compliance"
+    def _get_table_headers(self) -> list:
+        headers = DETECTIONS_COMMON_HEADERS.copy()
 
-                headers.extend(PREVIEW_DETECTIONS_COMMON_HEADERS)
-                headers.append(LICENSE_COLUMN)
+        if self._is_git_repository():
+            headers.insert(3, COMMIT_SHA_COLUMN)
 
-                for detection in detections:
-                    rows.append(self._get_license(detection))
+        if self.scan_type == SECRET_SCAN_TYPE:
+            headers.insert(3, SECRET_SHA_COLUMN)
+            headers.append(VIOLATION_LENGTH_COLUMN)
+            headers.append(VIOLATION_COLUMN)
 
-            if rows:
-                self._print_table_detections(detections, headers, rows, title)
+        return headers
 
-    def _get_table_headers(self) -> list:
-        if self._is_git_repository():
-            return [REPOSITORY_COLUMN]
+    def _get_table_columns_width(self, headers: List[ColumnInfo]) -> List[int]:
+        header_width_size_cols = []
+        for header in headers:
+            width_multiplier = 1
+            if self.scan_type == SECRET_SCAN_TYPE:
+                width_multiplier = header.width_secret
+            elif self.scan_type == INFRA_CONFIGURATION_SCAN_TYPE:
+                width_multiplier = header.width_iac
+            elif self.scan_type == SAST_SCAN_TYPE:
+                width_multiplier = header.width_sast
+
+            header_width_size_cols.append(len(header.name) * width_multiplier)
+
+        return header_width_size_cols
+
+    def _get_detection_row(self, detection: Detection, document: Document) -> List[str]:
+        return [
+            *self._get_detection_summary_fields(detection, document.path),
+            *self._get_detection_code_segment_fields(detection, document),
+        ]
 
-        return []
+    def _get_detection_summary_fields(self, detection: Detection, document_path: str) -> List[str]:
+        issue_type = detection.message
+        if self.scan_type == SECRET_SCAN_TYPE:
+            issue_type = detection.type
 
-    def _print_table_detections(
-            self, detections: List[Detection], headers: List[str], rows, title: str
-    ) -> None:
-        self._print_summary_issues(detections, title)
-        text_table = Texttable()
-        text_table.header(headers)
+        rows = [
```

Comment:
detection columns values, no?

Reply:
Do you mean var name? Yeah, could be better. I just did in in the same style as the previous table printer for sca